client/src/App.jsx

```diff
@@ -0,0 +1,121 @@
+import './styles.scss';
+import '@fortawesome/fontawesome-free/css/all.min.css';
+
+import React, { useContext, useEffect } from 'react';
+import { Link, NavLink, Route, Routes, useNavigate } from 'react-router-dom';
+import { HomePage } from './pages/HomePage';
+import { LoginPage } from './pages/LoginPage';
+import { RegistrationPage } from './pages/RegistrationPage';
+import { ProfilePage } from './pages/ProfilePage';
+import { AccountActivationPage } from './pages/AccountActivationPage';
+import { RequireAuth } from './components/RequireAuth';
+import { usePageError } from './hooks/usePageError';
+import { AuthContext } from './components/AuthContext';
+import { Loader } from './components/loader/Loader';
+import { ResetPasswordRequest } from './pages/ResetPasswordRequest';
+import { ChangePasswordPage } from './pages/ChangePasswordPage';
+import { NotFoundPage } from './pages/NotFoundPage';
+
+export const App = () => {
+  const navigate = useNavigate();
+  const [error, setError] = usePageError();
+  const { isChecked, user, logout, checkAuth } = useContext(AuthContext);
+
+  useEffect(() => {
+    checkAuth();
+  }, []);
+
+  if (!isChecked) {
+    return <Loader />;
+  }
+
+  const profileLink = user ? `/profile/change_name/${user.id}` : '/login';
```

Comment:
This line assumes that the `user` object always has an `id` property. Ensure that this is always the case or add a check to prevent potential errors if `user.id` is undefined.